src/middlewared/middlewared/plugins/tunables.py

```diff
@@ -25,212 +26,107 @@ class TunableService(CRUDService):
+
+    async def do_update(self, _id, data):
         """
         Update Tunable of `id`.
         """
-        old = await self.get_instance(id)
+        old = await self.get_instance(_id)
 
         new = old.copy()
         new.update(data)
+        if old == new:
+            # nothing updated so return early
+            return old
 
-        await self.clean(new, 'tunable_update', old=old)
-        await self.validate(new, 'tunable_update')
+        if (comment := data.get('comment', '').strip()) and comment != new['comment']:
+            new['comment'] = comment
 
-        await self.lower(new)
+        if not new['enabled']:
+            await self.middleware.run_in_thread(self.get_or_set, new['var'], new['orig_value'])
 
-        await self.middleware.call(
-            'datastore.update',
-            self._config.datastore,
-            id,
-            new,
-            {'prefix': self._config.datastore_prefix}
+        _id = await self.middleware.call(
+            'datastore.update', self._config.datastore, _id, new, {'prefix': self._config.datastore_prefix}
         )
+        await self.middleware.call('service.restart', 'sysctl')
+        return await self.get_instance(_id)
 
-        if old['type'] == 'SYSCTL' and old['var'] in self.__default_sysctl and (
-            old['var'] != new['var'] or old['type'] != new['type']
-        ):
-            default_value = self.__default_sysctl.pop(old['var'])
-            cp = await run(['sysctl', f'{old["var"]}={default_value}'], check=False, encoding='utf8')
-            if cp.returncode:
-                self.middleware.logger.error(
-                    'Failed to set sysctl %r -> %r : %s', old['var'], default_value, cp.stderr
-                )
-
-        await self.middleware.call('service.reload', new['type'])
-
-        return await self.get_instance(id)
-
-    async def do_delete(self, id):
+    async def do_delete(self, _id):
         """
         Delete Tunable of `id`.
         """
-        tunable = await self.get_instance(id)
-        await self.lower(tunable)
-        if tunable['type'] == 'sysctl':
-            # Restore the default value, if it is possible.
-            value_default = self.__default_sysctl.pop(tunable['var'], None)
-            if value_default:
-                cp = await run(['sysctl', f'{tunable["var"]}={value_default}'], check=False, encoding='utf8')
-                if cp.returncode:
-                    self.middleware.logger.error(
-                        'Failed to set sysctl %r -> %r : %s', tunable['var'], value_default, cp.stderr
-                    )
-
-        response = await self.middleware.call(
-            'datastore.delete',
-            self._config.datastore,
-            id
-        )
-
-        await self.middleware.call('service.reload', tunable['type'].lower())
-
-        return response
-
-    @private
-    async def lower(self, data):
-        data['type'] = data['type'].lower()
+        _id = await self.get_instance(_id)
```

Comment:
Maybe not use `_id` here as it's a complete dictionary and just causes confusion in the usage 😅